code/game/objects/machinery/overwatch.dm

```diff
@@ -1214,7 +1244,129 @@ GLOBAL_LIST_EMPTY(active_cas_targets)
-	return dat
+	return dat			
+
+/obj/machinery/computer/camera_advanced/overwatch/send_orders(datum/source, atom/object)
+	var/turf/target_turf = get_turf(object)
+	if (current_order)
+		if(COOLDOWN_CHECK(src, cooldown_order_cic))
+			COOLDOWN_START(src, cooldown_order_cic, ORDER_COOLDOWN)
+			switch (current_order)
+				if (ATTACK_ORDER)
+					send_attack_orders(target_turf)
+					return
+				if (DEFEND_ORDER)
+					send_defend_orders(target_turf)
+					return
+				if (RETREAT_ORDER)
+					send_retreat_orders(target_turf)
+					return
+		to_chat(usr, "<span class='warning'>Your last order was too recent</span>")
+		return
+	return
+
+	
+/obj/machinery/computer/camera_advanced/overwatch/proc/notify_marine(mob/living/marine, turf/target_turf) ///Send an order to that specific marine if it's on the right z level
+	if(marine.z == target_turf.z)
+		marine.playsound_local(marine, "sound/effects/CIC_order.ogg", 10, 1)
+		switch(current_order)
+			if (ATTACK_ORDER)
+				to_chat(marine,"<span class='ordercic'>Command is urging you to attack the ennemy at [target_turf.loc.name]</span>")
+				return
+			if (DEFEND_ORDER)
+				to_chat(marine,"<span class='ordercic'>Command is urging you to defend our positions in [target_turf.loc.name]</span>")
+				return
+			if (RETREAT_ORDER)
+				to_chat(marine,"<span class='ordercic'>Command is urging you retreat now from [target_turf.loc.name]</span>")
+				return
+		
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_attack_orders(turf/target_turf) ///Send an attack order to all marines
+	new /obj/effect/temp_visual/order/attack_order(target_turf)
+	var/obj/screen/arrow/arrow_hud
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD]
+	var/list/final_list = squad_hud.hudusers
+	final_list -= current_user //We don't want the eye to have an arrow, it's silly
+	for(var/hud_user in final_list)
+		arrow_hud = new /obj/screen/arrow/attack_order_arrow
+		arrow_hud.add_hud(hud_user, target_turf)
+		notify_marine(hud_user, target_turf)
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_retreat_orders(turf/target_turf) ///Send a retreat order
+	new /obj/effect/temp_visual/order/retreat_order(target_turf)
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD]
+	var/list/final_list = squad_hud.hudusers
+	final_list -= current_user
+	for(var/hud_user in final_list)
+		notify_marine(hud_user, target_turf)
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_defend_orders(turf/target_turf) ///Send a defend order
+	new /obj/effect/temp_visual/order/defend_order(target_turf)
+	var/obj/screen/arrow/arrow_hud
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD]
+	var/list/final_list = squad_hud.hudusers
+	final_list -= current_user
+	for(var/hud_user in squad_hud.hudusers)
+		arrow_hud = new /obj/screen/arrow/defend_order_arrow
+		arrow_hud.add_hud(hud_user, target_turf)
+		notify_marine(hud_user, target_turf)
```

Comment:
Make this a generic send order proc, using a selected order to know what it gotta do different from the others.

Reply:
done